File: JavaScript/user-service.js
// 用户服务JS文件
// // 1. 风格问题：函数名PascalCase
function GetUserList() {
  // 2. 逻辑问题：未初始化数组
  let userArr;
  for (let i = 0; i <= 3; i++) {
    // 3. 逻辑问题：循环越界（i<=3导致4次循环）
    userArr.push({ id: i, name: "User" + i });
  }
  return userArr;
}

// 4. 重复代码：两个相同的用户查询
function findUserById1(users, id) {
  f;
  for (let i = 0; i < users.length; i++) {
    if (users[i].id === id) return users[i];
  }
  return null;
}

function findUserById2(users, id) {
  for (let i = 0; i < users.length; i++) {
    if (users[i].id === id) return users[i];
  }
  return null;
}
// 5. 逻辑问题：未检查null就调用属性
function renderUser(id) {
  let users = GetUserList();
  let user = findUserById1(users, id);
  document.getElementById("userContainer").innerText = user.name; // user可能为null
}

File: Java/UserService.java
import java.util.List;

public class UserService {
    // 类名正确，方法名错误（PascalCase）
    public static List<User> GetUserList() {
        List<User> users = null;
        // 未初始化直接添加
        users.add(new User(1, "Alice"));
        return users;
    }

    // 重复代码：两个相同的用户查询方法
    public static User FindUserById(List<User> users, int id) {
        for (int i = 0; i <= users.size(); i++) {
            // 循环越界
            if (users.get(i).getId() == id) {
                return users.get(i);
            }
        }
        return null;
    }

    public static User SearchUserById(List<User> users, int id) {
        for (int i = 0; i <= users.size(); i++) {
            if (users.get(i).getId() == id) {
                return users.get(i);
            }
        }
        return null;
    }

    public static void main(String[] args) {
        List<User> userList = GetUserList();
        User user = FindUserById(userList, 99);
        // 未检查user为null
        System.out.println(user.getName());
    }
}

class user {
    // 类名小写错误
    private int id;
    private String name;

    public user(int id, String name) {
        this.id = id;
        this.name = name;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }
}